watson-developer-cloud/discovery-components
# fix: adds track event hooks

#### What do these changes do/fix?

Add track event hooks to call track event analytics in tooling.

Adds onChange callbacks to these components:

- SearchFacets (passes back synthetic event) 
- SearchInput (passes back search value `string`, value entered into search bar)
- SearchResults (passes back search value `string`  value of spelling suggestion)
- ResultsPagination (passes back object with page number and page size)

#### How do you test/verify these changes?

Requires `yarn link-components` command in tooling.

Companion Tooling PR: https://github.ibm.com/Watson-Discovery/discovery-tooling/pull/7655

The tooling PR has console logs with the event callback information, to verify the onChange callbacks are working.

Issue: https://github.ibm.com/Watson-Discovery/the-tooling-core-tracker/issues/1388
